test/integration/sip_test.go

```diff
@@ -376,10 +390,53 @@ func TestSIPJoinOpenRoom(t *testing.T) {
 		return dtmfIn == "4567"
 	}, 5*time.Second, time.Second/2)
 
+	go func() {
+		// TransferSIPParticipant is synchronous
+		_, err = lk.SIP.TransferSIPParticipant(context.Background(), &livekit.TransferSIPParticipantRequest{
+			RoomName:            "test-open",
+			ParticipantIdentity: "sip_" + clientNumber,
+			TransferTo:          "tel:" + transferNumber,
+		})
+		require.NoError(t, err)
+		close(transferDone)
+	}()
+
+	require.Eventually(t, func() bool {
+		dmu.Lock()
+		defer dmu.Unlock()
+
+		return referRequest != nil
+
+	}, 5*time.Second, time.Second/2)
+
+	require.Equal(t, sipgo.REFER, referRequest.Method)
+	transferTo := referRequest.GetHeader("Refer-To")
+	require.Equal(t, "tel:"+transferNumber, transferTo.Value())
+
+	time.Sleep(notifyIntervalDelay)
+	err = cli.SendNotify(referRequest, "SIP/2.0 100 Trying")
+	require.NoError(t, err)
+
+	time.Sleep(notifyIntervalDelay)
+	err = cli.SendNotify(referRequest, "SIP/2.0 200 OK")
+	require.NoError(t, err)
```

Comment:
This part could be in `onRefer` callback, then there will be no need for a mutex and shared variable (`referRequest`). Same for the second test.